jdaviz/components/toolbar_nested.py

```diff
@@ -53,13 +65,74 @@ def __init__(self, viewer, tools_nested, default_tool_priority=[]):
         self.default_tool_priority = default_tool_priority
         self._handle_default_tool()
 
+        for msg in (AddDataMessage, RemoveDataMessage, ViewerAddedMessage, ViewerRemovedMessage,
+                    SpectralMarksChangedMessage):
+            self.viewer.hub.subscribe(self, msg,
+                                      handler=lambda _: self._update_tool_visibilities())
+
+    def _is_visible(self, tool_id):
+        # tools can optionally implement self.is_visible(). If not NotImplementedError
+        # the tool will always be visible
+        return getattr(self.tools[tool_id], 'is_visible', lambda: True)()
```

Comment:
This is really hard to read. How about this?

```suggestion
        if hasattr(self.tools[tool_id], 'is_visible'):
            return self.tools[tool_id].is_visible()
        return True
```

Reply:
is the comment before not sufficient?  I am a fan of defaults in `getattr`, but if its unreadable can expand it.

Reply:
The trailing `()` could be easily missed by future devs.

Reply:
hmmm, what if I made all the `is_visible` properties instead (I used to have them take arguments, but then removed that)? 

Otherwise I can adopt your proposed changes if you think its worth the extra lines and call.

Reply:
You sure you won't try to re-introduce args in the future? Anyways, this is a style issue; I have no problem with the function behavior, so if you really don't think my concern applies, you can ignore it.

Reply:
that's why I left it as a method for now.  I'll adopt your changes to avoid any potential confusion.  Thanks!